public/app/features/alerting/unified/state/AlertingQueryRunner.ts

```diff
@@ -41,12 +42,24 @@ export class AlertingQueryRunner {
     return this.subject.asObservable();
   }
 
-  run(queries: AlertQuery[]) {
+  async run(queries: AlertQuery[]) {
     if (queries.length === 0) {
       const empty = initialState(queries, LoadingState.Done);
       return this.subject.next(empty);
     }
 
+    // do not execute if one more of the queries are not runnable,
+    // for example not completely configured
+    for (const query of queries) {
+      if (query.datasourceUid !== ExpressionDatasourceUID) {
```

Comment:
I think we can use `!isExpressionQuery` here as that will also check for `__expr__` types?